homeassistant/components/denonavr/device.py

```diff
@@ -0,0 +1,35 @@
+"""Representation of Denon AVR device entities."""
+
+from denonavr import DenonAVR
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_MODEL
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .config_flow import CONF_MANUFACTURER, CONF_TYPE, DOMAIN
+
+
+class DenonDeviceEntity(Entity):
+    """Representation of a Denon Device Entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        receiver: DenonAVR,
+        unique_id: str,
+        config_entry: ConfigEntry,
+    ) -> None:
+        """Initialize the device."""
+        self._attr_unique_id = unique_id
```

Comment:
I usually leave the unique id in the constructor of the platform entities.

Reply:
Each entity requires a unique id, so I don't see an advantage in doing this.